src/quest_manager/models.py

```diff
@@ -565,6 +565,13 @@ def expired(self):
         # utilize existing code in QuestQuerySet method not_expired()
         return not Quest.objects.filter(id=self.id).not_expired().exists()
 
+    def remove_as_prereq(self):
+        """
+        When a quest is archived, remove it from all prerequisite lists
+        """
+        if self.archived:
+            self.prereq_prereq.all().delete()
```

Comment:
How will this handle prerequisites that have an alternate OR prereq object?

i.e. Prereq objects that have "quest1 OR quest2", how does it handle this?

Reply:
Also, I feel like this should be a method in the IsAPrereq mixin, and it doesn't seem necessary to check archived here, that's already being checked before the method is called, and in the future we may want to call this method for other reasons (and on other objects that implement IsAPrereq)

Reply:
> Prereq objects that have "quest1 OR quest2", how does it handle this?

Can you give me another example? I'm not sure I understand what you mean here..

Reply:
Here's an example of a quest with an alternate prerequisite.  In the example its a Rank OR a Badge, but they could both be quests.

![image](https://github.com/bytedeck/bytedeck/assets/10604391/9dda9b21-d31f-4526-ba0f-1fbfd7f9dda8)

Which can be added in the prereq form as an "Alternate"

![image](https://github.com/bytedeck/bytedeck/assets/10604391/6e5a1852-9913-43ad-aef8-fde888feb380)


THe docstring for the class I think describes it fairly well?

```
class Prereq(IsAPrereqMixin, models.Model):
    """
    A Prereq object indicates some conditions (prerequisites) that must be met before gaining access to something else.

    parent_object: The parent object is the thing with restricted access (for example, a quest, or a badge).
     Access to it is granted once the conditions are met.

    prereq_object: This is the main condition that has to be met for a user to gain access to the parent object.
     How that condition is met is determined by the Class's implementation of `condition_met_as_prerequisite()` from the
     IsAPrereqMixin.  This object could be any model that implements the mixin (quest, badge, grade, course, etc)

    or_prereq_object: This is an ALTERNATE condition that could be met instead of the condition laid out by prereq_
     object.

    Simple example:  Imagine a quest that a student only gains access to if they are in a grade 10 course:

        Prereq <
            parent_object = quest_manager.models.Quest object <"Some Quest">
            prereq_object = courses.models.Grade <"Grade 10">
        >

        Given a specific user, when the querying of available quests gets to this quest, it will find this Prereq and
          call condition_met() on it. If condition_met() returns True (supposedly, because the student has a Grade 10
          course) then this Prereq (Quest object <"Some Quest"> will not be filtered out of the user's available quests

          See condition_met() for further details.


    Links two (or three) objects (of any registered content type) in a prerequisite relationship.
    Note that a Prereq can itself be a prereq (in combination with AND OR NOT, this allows for arbitrary complexity)
    parent: The owner of the prerequisite. Generally, for the parent to become 'active' the prereq_object conditions
        must be met.  'active' can mean whatever the user of this app wants it to mean.  It's a generic trigger.
    prereq: The object whose conditions must be met in order for the parent to become 'active', which optional
        additional conditions NOT and a number to indicate how many times the condition much be met.
    or_prereq: An optional alternate object who's condition can me met instead of prereq: prereq OR or_prereq
    """
```